tee-worker/omni-executor/intent/executors/cross-chain/src/lib.rs

```diff
@@ -1079,12 +1079,37 @@ fn str_to_u256(amount: &str, decimals: u32) -> Option<U256> {
 	U256::from_str(&int_str).ok()
 }
 
-fn calculate_amount_in(amount: &str, gas: &str) -> Option<String> {
+fn calculate_amount_in(amount: &str, gas: &str, decimals: u32) -> Option<String> {
 	let amount = Decimal::from_str(amount).ok()?;
 	let gas = Decimal::from_str(gas).ok()?;
 	if amount <= gas {
 		None
 	} else {
-		Some((amount - gas).to_string())
+		let amount = amount - gas;
+		Some(amount.trunc_with_scale(decimals).to_string())
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+
+	#[test]
+	fn test_str_to_u256() {
+		let amount = "0.0050773374896233926847036101";
+		let decimals = 18;
+
+		let result = str_to_u256(amount, decimals);
+		assert_eq!(result, Some(U256::from_str("5077337489623392").unwrap()));
+	}
+
+	#[test]
+	fn test_calculate_amount_in() {
+		let bnb_to_receive = "0.0050773374896233926847036101";
```

Comment:
Do you know from where do we get value with too many decimals ?

Reply:
The `amount_in` that we pass to backend => backend complains it doesn't match the token precision (e.g. > 18 decimal precision for `BNB`)